### PR TITLE
Update navbar usage and documentation

### PR DESCRIPTION
**Pull request by Chng-Zhi-Xuan**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] Documentation update
• [x] Enhancement to an existing feature

Part of #733 

**Note:** |
--- |
This PR is branched from #733. Thus the commits are shared with it. <br>First unique commit is titled `Asset: Update vue-strap.min.js`. |

**What changes did you make? (Give an overview)**
- Added updated `vue-strap.min.js` from MarkBind/vue-strap#101
- Updated `navBars.mbdf` to reflect new behaviour of navbars
- Updated `spacer-top` class
- Refactored usage of `header.md` within documentation and test_site to a header file
- Added header content test within test_site

**Is there anything you'd like reviewers to focus on?**
- Commit messages
- Documentation changes

**Testing instructions:**
- Browse documentation using Netlify Preview

### CODE REVIEW

File: asset/css/markbind.css [hunk 1]
```diff
@@ -116,11 +116,8 @@ footer {
 /* Side Navigation menus */
 
 .spacer-top {
-    /*
-      spacer-top class is a work around for MarkBind <navbar> component using "position: fixed".
-      This class can be removed once the component's position type is changed.
-    */
-    padding-top: 70px;
+    /* This class is to ensure adequate spacing of contents from the top of the page */
```
Comment (marvinchin) [inline asset/css/markbind.css]: Just wondering - should this comment be placed outside the class definition (even though the comment was previously inside the class definition)? 🙂

File: docs/userGuide/syntax/navBars.mbdf [hunk 2]
```diff
@@ -2,11 +2,17 @@
 
 **Navbar allows visitors of your website to navigate through pages easily.**
 
+<box type="warning">
+  <markdown>
+Note: **Navbars** should be placed within a [header file]({{ baseUrl }}/userguide/tweakingthepagestructure.html#headers) to ensure correct positioning at the top of your page.
```
Comment (marvinchin) [inline docs/userGuide/syntax/navBars.mbdf]: Should we change "to ensure correct positioning" -> "to ensure that they are correctly positioned"?

File: asset/css/markbind.css (hunk 1 above)
Comment (Chng-Zhi-Xuan) [reply asset/css/markbind.css]: For this section, how I formatted it was 

```CSS
/**
  * Main section description
  */

/* sub section one */

/* sub section two */

.some-class {
  /* class description */
}
```

My intention was to separate sub section title comments from class descriptions

File: docs/userGuide/syntax/navBars.mbdf (hunk 2 above)
Comment (Chng-Zhi-Xuan) [reply docs/userGuide/syntax/navBars.mbdf]: "... to ensure that they are correctly positioned above `page navigation` and `site navigation` menus." ?

Added more explicit interaction description.

Comment (luyangkenneth) [reply docs/userGuide/syntax/navBars.mbdf]: How about:

"... to ensure that they are correctly positioned at the top of the page, above the `page navigation` and `site navigation` menus."

**Comment by luyangkenneth** [comment]
Also, I'm not sure which specific PR introduced this change., but the "User Guide" heading in the site nav is wrongly indented.

**Comment by yamgent** [comment]
As spoken, need to rebase this PR to get rid of the commits that originated from #733.

**Comment by yamgent** [comment]
> Seems like the navbar is no longer fixed to the top of the page. Is this intentional?

@luyangkenneth As discussed with Prof, yes this is intentional, it follows the style of other websites such as [Jekyll's home page](https://jekyllrb.com/).

**Comment by damithc** [comment]
> @luyangkenneth As discussed with Prof, yes this is intentional, it follows the style of other websites such as [Jekyll's home page](https://jekyllrb.com/).

As I understood, there are technical difficulties in making the navbar fixed after the recent changes to add support for page headers. Some (but not all) other similar sites such as Jekyll seem to have non-fixed nav bars too. So, I don't mind dropping the fixed navbar until we can bring back support for fixed navbars.

**Comment by Chng-Zhi-Xuan** [comment]
**Update**

- Rebase to latest master
- Made minor documentation amendments, following review comments.

File: test/functional/test_site/_markbind/headers/header.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<header>
+  <navbar placement="top" type="inverse">
```
Comment (yamgent) [inline test/functional/test_site/_markbind/headers/header.md]: `placement="top"` is redundant here.

**Comment by Chng-Zhi-Xuan** [comment]
**Update**

- Rebase to latest master
- Removed `placement="top"` from `header.md` in test_site

**Pull request merged**